src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -145,33 +145,48 @@ async function has_bucket_policy_permission(policy, account, method, arn_path, r
     return 'IMPLICIT_DENY';
 }
 
-async function _is_statements_fit(statements, account, method, arn_path, req) {
-    for (const statement of statements) {
-        let action_fit = false;
-        let principal_fit = false;
-        let resource_fit = false;
-        let condition_fit = true;
-        for (const action of _.flatten([statement.Action])) {
-            dbg.log1('bucket_policy: action fit?', action, method);
-            if ((action === '*') || (action === 's3:*') || (action === method)) {
-                action_fit = true;
-            }
+function _is_action_fit(method, statement_action, isNotStatement) {
```

Comment:
(related also to `_is_principal_fit`, `_is_resource_fit`):
1. Would you please calculate the `isNotStatement` inside the function itself?
It would be more readable in statements calling this function.
2. Would you please add a comment with a short explanation of what we check?
Specially in `_is_resource_fit` with the regular expression.